content/en/blog/_posts/2022-08-15-local-storage-capacity-isolation-ga.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: blog
+title: "Kubernetes 1.25: Local Storage Capacity Isolation Reaches GA"
+date: 2022-08-15
+slug: local-storage-capacity-isolation-ga
+---
+
+**Author:** Jing Xu (Google)
+
+Local ephemeral storage capacity isolation was introduced as a alpha feature in Kubernetes 1.7 and it went beta in 1.9 and with Kubernetes 1.25 we are excited to announce general availability(GA) of this feature.
+
+Pods use ephemeral local storage for scratch space, caching, and for logs. The lifetime of local ephemeral storage does not extend beyond the life of the individual pod. It is exposed to pods using the container’s writable layer, logs directory, and EmptyDir volumes. Before this feature available, there have been issues related to the lack of local storage accounting and isolation, such as Pods do not know how much local storage is available and cannot request guaranteed local storage. Local storage is a best effort resource and pods can be evicted due to other pods filling the local storage.
+
+The local storage capacity isolation feature allows users to manage local ephemeral storage in the same way as managing CPU and memory. It provides support for capacity isolation of shared storage between pods, such that a pod can be hard limited in its consumption of shared resources by evicting Pods if its consumption of shared storage exceeds that limit. It also allows setting ephemeral storage request for resource reservation. The limits and requests for shared ephemeral-storage are similar to those for memory and CPU consumption.
+
+
+
+### How to use local storage capacity isolation
+
+A typical configuration for local ephemeral storage is to place all different kinds of ephemeral local data (emptyDir volumes, writeable layers, container images, logs) into one filesystem. Typically, both /var/lib/kubelet and /var/log are on the system root filesystem. If users configure the local storage in different ways, kubelet might not be able to correctly measure disk usage and use this feature.
+
+
+#### Setting requests and limits for local ephemeral storage
+You can specify `ephemeral-storage` for managing local ephemeral storage. Each container of a Pod can specify either or both of the following:
+
+* `spec.containers[].resources.limits.ephemeral-storage`
+* `spec.containers[].resources.requests.ephemeral-storage`
+
+In the following example, the Pod has two containers. The first container has a request of 2GiB of local ephemeral storage and a limit of 5GiB. The second container requests 6GiB of local storage. Therefore, the Pod has a request of total 10GiB (8GiB+2GiB) of local ephemeral storage, and a limit of 6GiB of local ephemeral storage. It also sets emptyDir sizeLimit to 5GiB. With this setting in pod spec, it will affect how scheulder makes a decision on scheduling pods and also how kubelet evict pods.
```

Comment:
I'm not sure of the math here. I could be wrong, conceptually, but if the first container requests 2 GiB & the second container requests 6 GiB shouldn't the total requested be 8 GiB? We either need to change the example or the math.

```suggestion
In the following example, the Pod has two containers. The first container has a request of 2GiB of local ephemeral storage and a limit of 5GiB. The second container requests 6GiB of local storage. Therefore, the Pod requests a total of 10GiB (8GiB+2GiB) of local ephemeral storage and enforces a limit of 6GiB of local ephemeral storage. It also sets emptyDir sizeLimit to 5GiB. With this setting in pod spec, it will affect how scheulder makes a decision on scheduling pods and also how kubelet evict pods.
```